### PR TITLE
Document copy-on-write arrays

### LINKED ISSUE #695: Array docs should mention copy-on-write
It's currently not clear from this page that vec/dicts/keysets/arrays are copy-on-write.

--------------------------------

Please don't change anything below this point.

--------------------------------

 - Build ID: HHVM=HHVM-4.15.0:HSL=v4.7.0:2019-07-31T15:43:43+0000:982a9335b437e2a9b53d8bbdf6c9d78c5609b75b
 - Page requested: /hack/built-in-types/arrays
 - Page requested at: Thu, 01 Aug 2019 10:47:07 +0000
 - Controller: GuidePageController

### PR DESCRIPTION
**Pull request by lexidor**
closes #695

### CODE REVIEW

**Comment by lexidor** [comment]
Superceeds #713

File: guides/hack/11-built-in-types/22-arrays-examples/assignment-with-value-containers.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?hh // strict
+
+namespace Hack\UserDocumentation\Types\Arrays\Examples\AssignmentWithValueContainers;
+
+use namespace HH\Lib\Str;
+
+<<__EntryPoint>>
+function main(): void {
+    require_once __DIR__.'/../../../../vendor/autoload.hack';
+    \Facebook\AutoloadMap\initialize();
```
Comment (fredemmott) [inline guides/hack/11-built-in-types/22-arrays-examples/assignment-with-value-containers.php]: This should be unneeded - the test runner automatically pulls this in

Comment (lexidor) [reply guides/hack/11-built-in-types/22-arrays-examples/assignment-with-value-containers.php]: So, my `<<__EntryPoint>>` isn't the _real_ entrypoint.

Comment (fredemmott) [reply guides/hack/11-built-in-types/22-arrays-examples/assignment-with-value-containers.php]: well... kinda - `auto_prepend_file` ini setting is still a thing :p

**Pull request merged**